presto-main/src/main/java/com/facebook/presto/security/AccessControl.java

```diff
@@ -165,18 +167,29 @@
     void checkCanCreateViewWithSelectFromView(TransactionId transactionId, Identity identity, QualifiedObjectName viewName);
 
     /**
-     * Check if identity is allowed to grant a privilege on the specified table.
+     * Check if identity is allowed to grant the specified privilege on the specified table.
      * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
      */
     void checkCanGrantTablePrivilege(TransactionId transactionId, Identity identity, Privilege privilege, QualifiedObjectName tableName);
 
     /**
-     * Check if identity is allowed to revoke a privilege on the specified table.
+     * Check if identity is allowed to revoke the specified privilege on the specified table.
      * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
      */
     void checkCanRevokeTablePrivilege(TransactionId transactionId, Identity identity, Privilege privilege, QualifiedObjectName tableName);
 
     /**
+     * Check if identity is allowed to show grants on the specified catalog or schema or table.
+     * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
+     */
+    void checkCanShowGrants(TransactionId transactionId, Identity identity, QualifiedTablePrefix prefix);
```

Comment:
Ok, I could have reused `checkCanShowTables` method (perhaps not `filterTables` method, since what is being returned is different than in `filterGrants`). But then, we have to think about what error message to throw. `checkCanShowTables` throws "Cannot show tables..." message, which won't be right in this case. One option is to create a common method `checkCanShowTableMetadata` that will throw an error message along the lines of "Cannot show metadata for tables in ...". So considering the refactoring involved, I didn't change the access control methods for now. Thoughts?